src/pydvl/utils/parallel/backend.py

```diff
@@ -169,9 +171,13 @@ def wrap(self, fun: Callable, **kwargs) -> Callable:
 
         :return: The `.remote` method of the ray `RemoteFunction`.
         """
-        if len(kwargs) > 1:
-            return ray.remote(**kwargs)(fun).remote  # type: ignore
-        return ray.remote(fun).remote  # type: ignore
+        remote_fn = ray.remote(fun)
+        if len(kwargs) == 0:
+            kwargs["num_cpus"] = self.n_cpus_per_job
+        else:
+            if "num_cpus" not in kwargs:
+                kwargs["num_cpus"] = self.n_cpus_per_job
+        return remote_fn.remote  # type: ignore
```

Comment:
This seems to be a mistake? With this change, kwargs are no longer passed to `ray.remote()`
Also, see my comment below about `n_cpus_per_job`, and I don't think it belongs in the instance's state. If it isn't set by the user in the kwargs, that's it.
The previous code takes into account the 2 different ways of calling ray.remote, I think it should be left as it was.
```suggestion
        if len(kwargs) > 1:
            return ray.remote(**kwargs)(fun).remote  # type: ignore
        return ray.remote(fun).remote  # type: ignore
```